corehq/apps/data_dictionary/tests/test_import_export.py

```diff
@@ -0,0 +1,133 @@
+import io
+import os
+import uuid
+
+from bs4 import BeautifulSoup
+import openpyxl
+
+from django.test import TestCase
+from django.urls import reverse
+
+from corehq.apps.data_dictionary.models import CaseProperty, CasePropertyAllowedValue, CaseType
+from corehq.apps.data_dictionary.views import ExportDataDictionaryView, UploadDataDictionaryView
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.users.models import WebUser
+from corehq.util.test_utils import flag_enabled
+
+
+@flag_enabled('DATA_DICTIONARY')
+class DataDictionaryImportTest(TestCase):
+    domain_name = uuid.uuid4().hex
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = create_domain(cls.domain_name)
+        cls.couch_user = WebUser.create(None, "test", "foobar", None, None)
+        cls.couch_user.add_domain_membership(cls.domain_name, is_admin=True)
+        cls.couch_user.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.couch_user.delete(cls.domain_name, deleted_by=None)
+        cls.domain.delete()
+        super().tearDownClass()
+
+    def setUp(self):
+        self.url = reverse(UploadDataDictionaryView.urlname, args=[self.domain_name])
+        self.client.login(username='test', password='foobar')
+
+    def test_clean_import(self):
+        fname = os.path.join(os.path.dirname(__file__), 'data', 'clean_data_dictionary.xlsx')
```

Comment:
There's a util to do this: the test class can use the mixin `TestFileMixin` and that includes a `get_file` helper (based on a `file_path` attribute of the test class. Check out [WorkbookJSONReaderTest](https://github.com/dimagi/commcare-hq/blob/d127f07ceb000d6647a9003d46d337f35220a54f/corehq/util/tests/test_excel.py#L14) for an example.